pkg/driver/common/operator/hooks.go

```diff
@@ -35,21 +36,29 @@ func NewDefaultOperatorControllerConfig(flavour generator.ClusterFlavour, c *cli
 	}
 	// Default controller hooks
 	if flavour == generator.FlavourStandalone {
-		cfg.AddDeploymentHookBuilders(c, withClusterWideProxy, withStandaloneReplicas)
+		cfg.AddDeploymentHookBuilders(c, withClusterWideProxyDeploymentHook, withStandaloneReplicas)
 	} else {
 		// HyperShift
 		cfg.AddDeploymentHookBuilders(c, withHyperShiftReplicas, withHyperShiftNodeSelector, withHyperShiftControlPlaneImages)
 	}
 
+	cfg.AddDaemonSetHookBuilders(c, withClusterWideProxyDaemonSetHook)
+
 	return cfg
 }
 
-// withClusterWideProxy adds the cluster-wide proxy config to the Deployment.
-func withClusterWideProxy(c *clients.Clients) (dc.DeploymentHookFunc, []factory.Informer) {
+// withClusterWideProxyDeploymentHook adds the cluster-wide proxy config to the Deployment.
+func withClusterWideProxyDeploymentHook(c *clients.Clients) (dc.DeploymentHookFunc, []factory.Informer) {
 	hook := csidrivercontrollerservicecontroller.WithObservedProxyDeploymentHook()
 	return hook, nil
 }
 
+// withClusterWideProxyHook adds the cluster-wide proxy config to the DaemonSet.
+func withClusterWideProxyDaemonSetHook(_ *clients.Clients) (csidrivernodeservicecontroller.DaemonSetHookFunc, []factory.Informer) {
```

Comment:
So, this hook is not needed for all CSI drivers because for EFS and EBS drivers, usually they don't talk to cloud on the node side at all (I could be wrong about this for EFS, will have to double check). So, that is why you don't see this applied for every driver in this package. 

IMO - I will keep this hook on per-driver basis for daemonset because not every driver needs this.

Reply:
Okay. As I noted above though, [both the EBS and EFS drivers have the `config.openshift.io/inject-proxy` annotation](https://github.com/openshift/csi-operator/pull/304#issuecomment-2422834547). I can drop the second commit here, but should I replace it with one that drops those annotations since they're apparently not needed?

Reply:
I've dropped this from this PR and will move to a separate PR.

Reply:
PS: Separate PR is [here](https://github.com/openshift/csi-operator/pull/308).